src/extensions/mountainview/MVSortingView/PreloadCheck.tsx

```diff
@@ -0,0 +1,115 @@
+                        setMessage('Downloading sorting...')
+                        const result2 = await hither.createHitherJob('preload_download_sorting', {sorting_object: sortingObject}, {useClientCache: false, calculationPool}).wait() as {success: boolean}
+                        if (!matchesRunningState({recordingObject, sortingObject})) return
+                        if (!result2.success) {
+                            setError(new Error('Unable to download sorting.'))
+                            return
+                        }
+                    }
+
+                    setMessage('Checking recording local...')
+                    const result3 = await hither.createHitherJob('preload_check_recording_downloaded', {recording_object: recordingObject}, {useClientCache: false, calculationPool}).wait() as {isLocal: boolean}
+                    if (!matchesRunningState({recordingObject, sortingObject})) return
+                    if (!result3.isLocal) {
+                        setMessage('Downloading recording...')
+                        const result4 = await hither.createHitherJob('preload_download_recording', {recording_object: recordingObject}, {useClientCache: false, calculationPool}).wait() as {success: boolean}
+                        if (!matchesRunningState({recordingObject, sortingObject})) return
+                        if (!result4.success) {
+                            setError(new Error('Unable to download recording.'))
+                            return
+                        }
+                    }
+
+                    setMessage('Extracting snippets...')
+                    const result5 = await hither.createHitherJob('preload_extract_snippets', {recording_object: recordingObject, sorting_object: sortingObject}, {useClientCache: false, calculationPool}).wait() as string
+                    if (!matchesRunningState({recordingObject, sortingObject})) return
+                    if (!result5) {
+                        setError(new Error('Problem extracting snippets'))
+                        return
+                    }
+                    setStatus('finished')
+                }
+                catch(err) {
+                    setError(err)
+                }
+            })()
+        }
+        else {
+            if (!matchesRunningState({recordingObject, sortingObject})) {
+                setStatus('waiting')
+            }
+        }
+    }, [sortingObject, recordingObject, status, matchesRunningState, hither])
+
+    const child = useMemo(() => {
+        return React.cloneElement(
+            children,
+            {
+                preloadStatus: status
+            }
+        )
+    }, [children, status])
+
+    // This is important for when the bandpass filter changes so that we don't start calculating things prior to doing the preloading (e.g. snippets extraction)
+    const [lastValidChild, setLastValidChild] = useState<React.ReactElement | null>(null)
```

Comment:
This is important for when the bandpass filter changes so that we don't start calculating things prior to doing the preloading (e.g. snippets extraction)